### PR TITLE
SUP-1861: Some more CircleCI examples 

### PR DESCRIPTION
**Pull request by james2791**
More Circle examples - shell setting in runs in job steps, parallelism, top-level parameters, executor definitions, Java / RoR pipelines

### CODE REVIEW

File: app/spec/lib/bk/compat/circleci/examples/workflow-dependencies.yml
Comment (toote) [inline app/spec/lib/bk/compat/circleci/examples/workflow-dependencies.yml:1]: good catch, I didn't realize we were not testing this

**Review by toote — APPROVED** [review]
just a minor issue with one CircleCI file but great addition!

File: app/spec/lib/bk/compat/circleci/__snapshots__/spec/lib/bk/compat/circleci/examples/java-maven.yml.snap [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+steps:
+- commands:
+  - "# No need for checkout, the agent takes care of that"
+  - mvn clean install
+  - mvn test
+  - mvn package
+  - cd ~/project
```
[the remark this answers is not in the document]
Comment (james2791) [reply app/spec/lib/bk/compat/circleci/__snapshots__/spec/lib/bk/compat/circleci/examples/java-maven.yml.snap:8]: Ticketing <3

File: app/spec/lib/bk/compat/circleci/examples/executor-top-level.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+ersion: 2.1
```
[the remark this answers is not in the document]
Comment (james2791) [reply app/spec/lib/bk/compat/circleci/examples/executor-top-level.yml:1]: Missed this bugger - corrected in latest

File: app/spec/lib/bk/compat/circleci/examples/workflow-dependencies.yml
Comment (james2791) [reply app/spec/lib/bk/compat/circleci/examples/workflow-dependencies.yml:1]: Relatively simple example but testing all the things™

**Pull request merged**